Introduce word counter

shtukas opened the pull request:
... in the toolbar. 

Note:
- It wasn't possible to code this feature in scala(js), because putting an event handler on
  `bodyTag`, for instance `onkeypress`, didn't work. So I implemented in JavaScript.
- Then a jQuery `$( "#explainer-editor__body-wrapper__input" ).keypress` didn't work either.

Working solution was a

```
setInterval(function(){updateWordCountDisplay();},100);
```

shtukas committed:
Introduce word counter

In the toolbar. Note:

- It wasn't possible to code this feature in scala(js), because putting an event handler on
  `bodyTag`, for instance `onkeypress`, didn't work. So I implemented in JavaScript.

- Then a jQuery `$( "#explainer-editor__body-wrapper__input" ).keypress` didn't work either.

Working solution was a

```
setInterval(function(){updateWordCountDisplay();},100);
```

shtukas committed:
small refactoring

shtukas commented:
Deployed and tested on CODE.

shtukas commented:
![picture 3](https://cloud.githubusercontent.com/assets/6035518/16997576/f7589d24-4ead-11e6-98b5-f62192d7b851.png)

shtukas commented:
Note: The warning(s) that the word count is too high will come is a separate request.

philmcmahon commented on explainer-server/app/views/templates/main.scala.html:
```diff
@@ -42,6 +42,21 @@
+         * #explainer-editor__body-wrapper__input
+         * This explains the choice of setInterval
+         */
+        function updateWordCountDisplay(){
```
This looks like a plain javascript function now. Could you write it in scala js?

shtukas replied:
As discussed, the event handlers could not work in either scalajs or plain javascript, so I moved to setInterval in JavaScript, but I can try that in scalajs :)

philmcmahon commented on explainer-server/app/views/templates/main.scala.html:
```diff
@@ -42,6 +42,21 @@
+            var wordCount = contents.split(" ").length;
+            $("#top-toolbar__word-count-wrapper__count").text(wordCount)
+        }
+        setInterval(function(){
```
setInterval is also possible in scala js?

shaundillon commented on explainer-server/public/stylesheets/main.css:
```diff
@@ -212,5 +212,11 @@ body {
 
+#top-toolbar__word-count-wrapper {
+  border-left: 1px solid #bdbdbd;
+  padding:0px 15px 0px 15px;
```
This can be done in shorthand, the `bottom` and `left` will inherit from `top` and `right` respectively. Also, you shouldn't use units on 0 values as it's not needed.
So you can change this to `padding: 0 15px` for the same effect.

shtukas replied:
Done. Thanks!

shaundillon commented on explainer-server/public/stylesheets/main.css:
```diff
@@ -212,5 +212,11 @@ body {
+}
 
-
+#top-toolbar__word-count-wrapper__count {
```
Try not to use id selectors in CSS. The specificity is too high and causes problems later on.

shtukas replied:
Done!

shtukas committed:
Moving the word count JS code from the main template to the Editor Initialization code

shtukas committed:
Moving from id to class in counter reference

shtukas committed:
CSS correction

shtukas committed:
Handler is in scalajs but called function is pure scala :)

shaundillon commented on explainer-server/public/stylesheets/main.css:
```diff
@@ -212,5 +212,11 @@ body {
   border-left: 1px solid #bdbdbd;
 }
 
+#top-toolbar__word-count-wrapper {
```
Still using an ID here haha

shtukas replied:
Yes, sorry I changed the other one but not this one...

shtukas committed:
comments

piuccio commented on explainer-server/app/views/explainEditor.scala.html:
```diff
@@ -25,5 +28,18 @@
+         * the read operation of the text area content uses cached values in the JS code generated by ScalaJS.
+         */
+        function updateWordCountDisplay(){
+            var contents = $.trim($("#explainer-editor__body-wrapper__input").val())
```
in general it's better avoiding ids.

ids are not future compatible if you ever want to change the UI and have two explainer in a page

shtukas replied:
Yes, Shaun was saying the same thing...

shtukas replied:
Done!

shtukas committed:
Word Count updates

1. Moving from CSS id to CSS class for top-toolbar__word-count-wrapper.
2. Enablying the display of the word count at page load.

shtukas committed:
Moving from CSS id to CSS class for explainer-editor__body-wrapper__input

shaundillon commented:
👍
